### PR TITLE
[Localization] Add localization hard-coded message in move, pokemon.

### PR DESCRIPTION
**Pull request by EnochG1**
## What are the changes the user will see?
English : none.
Other languages : Can see translated messages.

## Why am I making these changes?
All message texts are localizable.

## What are the changes from a developer perspective?
- Change hard-coded messages in `src/data/move.ts`, `src/field/pokemon.ts`, `src/phases/move-phase.ts`
```Typescript
this.scene.queueMessage(`${this.name} is unaffected!`);
this.scene.queueMessage(`${pokemon.name} was revived!`,0,true);
user.scene.queueMessage(`${getPokemonNameWithAffix(pokemon)} was revived!`,0,true);
this.scene.queueMessage(`${this.move.getName()} is disabled!`);
```
- Add locale texts in each language locale files.

#### En
<details>
<summary>Click to expand/collapse</summary>

| Case | Result |
| :-----: | :-----: |
| Immune | image |
| Disabled | image |
| RevivalBlessing | image |

</details>

#### Es
<details>
<summary>Click to expand/collapse</summary>

| Case | Result |
| :-----: | :-----: |
| Immune | image |
| Disabled | image |
| RevivalBlessing | image |

</details>

#### It
<details>
<summary>Click to expand/collapse</summary>

| Case | Result |
| :-----: | :-----: |
| Immune | image |
| Disabled | image |
| RevivalBlessing | image |

</details>

#### Fr
<details>
<summary>Click to expand/collapse</summary>

| Case | Result |
| :-----: | :-----: |
| Immune | image |
| Disabled | image |
| RevivalBlessing | image |

</details>

#### De
<details>
<summary>Click to expand/collapse</summary>

| Case | Result |
| :-----: | :-----: |
| Immune | image |
| Disabled | image |
| RevivalBlessing | image |

</details>

#### Pt_BR
<details>
<summary>Click to expand/collapse</summary>

| Case | Result |
| :-----: | :-----: |
| Immune | image |
| Disabled | image |
| RevivalBlessing | image |

</details>

#### Zh_CN
<details>
<summary>Click to expand/collapse</summary>

| Case | Result |
| :-----: | :-----: |
| Immune | image |
| Disabled | image |
| RevivalBlessing | image |

</details>

#### Zh_TW
<details>
<summary>Click to expand/collapse</summary>

| Case | Result |
| :-----: | :-----: |
| Immune | image |
| Disabled | image |
| RevivalBlessing | image |

</details>

#### Ko
<details>
<summary>Click to expand/collapse</summary>

| Case | Result |
| :-----: | :-----: |
| Immune | image |
| Disabled | image |
| RevivalBlessing | image |

</details>

#### Ja
<details>
<summary>Click to expand/collapse</summary>

| Case | Result |
| :-----: | :-----: |
| Immune | image |
| Disabled | image |
| RevivalBlessing | image |

</details>

## How to test the changes?
Manually. (Using overrides.ts)
1. ${this.name} is unaffected! > test with Moves.SHEER_COLD to ice type pokemon.
2. ${this.move.getName()} is disabled! > test with Moves.DISABLE
3. ${pokemon.name} was revived! > test with Moves.REVIVAL_BLESSING

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [ ] Have I considered writing automated tests for the issue?
- [x] If I have text, did I add placeholders for them in locales?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/locales/de/battle.ts [hunk 1]
```diff
@@ -24,6 +24,7 @@ export const battle: SimpleTranslationEntries = {
   "hitResultSuperEffective": "Das ist sehr effektiv!",
   "hitResultNotVeryEffective": "Das ist nicht sehr effektiv…",
   "hitResultNoEffect": "Es hat keine Wirkung auf {{pokemonName}}…",
+  "hitResultImmune": "{{pokemonName}} is unaffected!",
```
Comment (CodeTappert) [inline src/locales/de/battle.ts:27]:
```suggestion
  "hitResultImmune": "{{pokemonName}} ist unversehrt!",
```

File: src/locales/de/move-trigger.ts [hunk 2]
```diff
@@ -61,6 +61,7 @@ export const moveTriggers: SimpleTranslationEntries = {
   "faintCountdown": "{{pokemonName}} geht nach {{turnCount}} Runden K.O.!",
   "copyType": "{{pokemonName}} hat den Typ von {{targetPokemonName}} angenommen!",
   "suppressAbilities": "Die Fähigkeit von {{pokemonName}} wirkt nicht mehr!",
+  "revivalBlessing": "{{pokemonName}} was revived!",
```
Comment (CodeTappert) [inline src/locales/de/move-trigger.ts:64]:
```suggestion
  "revivalBlessing": "{{pokemonName}} ist wieder fit!",
```

File: src/locales/zh_TW/move-trigger.ts [hunk 3]
```diff
@@ -61,6 +61,7 @@ export const moveTriggers: SimpleTranslationEntries = {
   "faintCountdown": "{{pokemonName}}\n將在{{turnCount}}回合後滅亡！",
   "copyType": "{{pokemonName}}變成了{{targetPokemonName}}的屬性！",
   "suppressAbilities": "{{pokemonName}}的特性\n變得無效了！",
+  "revivalBlessing": "{{pokemonName}} was revived!",
```
Comment (mercurius-00) [inline src/locales/zh_TW/move-trigger.ts:64]:
```suggestion
  "revivalBlessing": "{{pokemonName}}復活了！",
```

File: src/locales/zh_TW/battle.ts [hunk 4]
```diff
@@ -21,6 +21,7 @@ export const battle: SimpleTranslationEntries = {
   "hitResultSuperEffective": "效果拔群！",
   "hitResultNotVeryEffective": "收效甚微…",
   "hitResultNoEffect": "對 {{pokemonName}} 沒有效果！",
+  "hitResultImmune": "{{pokemonName}} is unaffected!",
```
Comment (mercurius-00) [inline src/locales/zh_TW/battle.ts:24]:
```suggestion
  "hitResultImmune": "對于{{pokemonName}}，\n完全沒有效果！",
```

File: src/locales/zh_CN/move-trigger.ts [hunk 5]
```diff
@@ -61,6 +61,7 @@ export const moveTriggers: SimpleTranslationEntries = {
   "faintCountdown": "{{pokemonName}}\n将在{{turnCount}}回合后灭亡！",
   "copyType": "{{pokemonName}}\n变成了{{targetPokemonName}}的属性！",
   "suppressAbilities": "{{pokemonName}}的特性\n变得无效了！",
+  "revivalBlessing": "{{pokemonName}} was revived!",
```
Comment (mercurius-00) [inline src/locales/zh_CN/move-trigger.ts:64]:
```suggestion
  "revivalBlessing": "{{pokemonName}}复活了！",
```

File: src/locales/zh_CN/battle.ts [hunk 6]
```diff
@@ -24,6 +24,7 @@ export const battle: SimpleTranslationEntries = {
   "hitResultSuperEffective": "效果拔群！",
   "hitResultNotVeryEffective": "收效甚微…",
   "hitResultNoEffect": "对{{pokemonName}}没有效果！！",
+  "hitResultImmune": "{{pokemonName}} is unaffected!",
```
Comment (mercurius-00) [inline src/locales/zh_CN/battle.ts:27]:
```suggestion
  "hitResultImmune": "对于{{pokemonName}}，\n完全没有效果！",
```

**Review by mercurius-00 — COMMENTED** [review]
zh

File: src/locales/it/battle.ts [hunk 7]
```diff
@@ -24,6 +24,7 @@ export const battle: SimpleTranslationEntries = {
   "hitResultSuperEffective": "È superefficace!",
   "hitResultNotVeryEffective": "Non è molto efficace…",
   "hitResultNoEffect": "Non ha effetto su {{pokemonName}}!",
+  "hitResultImmune": "{{pokemonName}} is unaffected!",
```
Comment (NicusPulcis) [inline src/locales/it/battle.ts:27]:
```suggestion
  "hitResultImmune": "{{pokemonName}} è incolume!",
```

File: src/locales/it/move-trigger.ts [hunk 8]
```diff
@@ -61,6 +61,7 @@ export const moveTriggers: SimpleTranslationEntries = {
   "faintCountdown": "{{pokemonName}}\nandrà KO dopo {{turnCount}} turni.",
   "copyType": "{{pokemonName}} assume il tipo\ndi {{targetPokemonName}}!",
   "suppressAbilities": "L’abilità di {{pokemonName}}\nperde ogni efficacia!",
+  "revivalBlessing": "{{pokemonName}} was revived!",
```
Comment (NicusPulcis) [inline src/locales/it/move-trigger.ts:64]:
```suggestion
  "revivalBlessing": "{{pokemonName}} torna in forze!",
```

**Review by NicusPulcis — APPROVED** [review]
italian localizations done : )

File: src/locales/fr/move-trigger.ts [hunk 9]
```diff
@@ -61,6 +61,7 @@ export const moveTriggers: SimpleTranslationEntries = {
   "faintCountdown": "{{pokemonName}}\nsera K.O. dans {{turnCount}} tours !",
   "copyType": "{{pokemonName}} prend le type\nde {{targetPokemonName}} !",
   "suppressAbilities": "Le talent de {{pokemonName}}\na été rendu inactif !",
+  "revivalBlessing": "{{pokemonName}} was revived!",
```
Comment (Adri1) [inline src/locales/fr/move-trigger.ts:64]:
```suggestion
  "revivalBlessing": "{{pokemonName}} a repris connaissance\net est prêt à se battre de nouveau !",
```

File: src/locales/fr/battle.ts [hunk 10]
```diff
@@ -24,6 +24,7 @@ export const battle: SimpleTranslationEntries = {
   "hitResultSuperEffective": "C’est super efficace !",
   "hitResultNotVeryEffective": "Ce n’est pas très efficace…",
   "hitResultNoEffect": "Ça n’affecte pas {{pokemonName}}…",
+  "hitResultImmune": "{{pokemonName}} is unaffected!",
```
Comment (Adri1) [inline src/locales/fr/battle.ts:27]:
```suggestion
  "hitResultImmune": "{{pokemonName}} n’est pas affecté !",
```

**Review by Adri1 — APPROVED** [review]

File: src/locales/pt_BR/battle.ts [hunk 11]
```diff
@@ -24,6 +24,7 @@ export const battle: SimpleTranslationEntries = {
   "hitResultSuperEffective": "É supereficaz!",
   "hitResultNotVeryEffective": "É pouco eficaz...",
   "hitResultNoEffect": "Isso não afeta {{pokemonName}}!",
+  "hitResultImmune": "{{pokemonName}} is unaffected!",
```
Comment (josericardo-fo) [inline src/locales/pt_BR/battle.ts:27]:
```suggestion
  "hitResultImmune": "{{pokemonName}} não é afetado!",
```

File: src/locales/pt_BR/move-trigger.ts [hunk 12]
```diff
@@ -61,6 +61,7 @@ export const moveTriggers: SimpleTranslationEntries = {
   "faintCountdown": "{{pokemonName}}\nirá desmaiar em {{turnCount}} turnos.",
   "copyType": "O tipo de {{pokemonName}}\nmudou para combinar com {{targetPokemonName}}!",
   "suppressAbilities": "A habilidade de {{pokemonName}}\nfoi suprimida!",
+  "revivalBlessing": "{{pokemonName}} was revived!",
```
Comment (josericardo-fo) [inline src/locales/pt_BR/move-trigger.ts:64]:
```suggestion
  "revivalBlessing": "{{pokemonName}} foi reanimado!",
```

File: src/locales/es/move-trigger.ts [hunk 13]
```diff
@@ -61,6 +61,7 @@ export const moveTriggers: SimpleTranslationEntries = {
   "faintCountdown": "{{pokemonName}}\nwill faint in {{turnCount}} turns.",
   "copyType": "{{pokemonName}}'s type\nchanged to match {{targetPokemonName}}'s!",
   "suppressAbilities": "{{pokemonName}}'s ability\nwas suppressed!",
+  "revivalBlessing": "{{pokemonName}} was revived!",
```
Comment (AsdarDevelops) [inline src/locales/es/move-trigger.ts:64]:
```suggestion
  "revivalBlessing": "¡{{pokemonName}} ha revivido!",
```

**Review by AsdarDevelops — COMMENTED** [review]
Added Spanish

File: src/locales/es/battle.ts [hunk 14]
```diff
@@ -24,6 +24,7 @@ export const battle: SimpleTranslationEntries = {
   "hitResultSuperEffective": "¡Es supereficaz!",
   "hitResultNotVeryEffective": "No es muy eficaz…",
   "hitResultNoEffect": "No afecta a {{pokemonName}}!",
+  "hitResultImmune": "{{pokemonName}} is unaffected!",
```
Comment (AsdarDevelops) [inline src/locales/es/battle.ts:27]:
```suggestion
  "hitResultImmune": "¡No afecta a {{pokemonName}}!",
```

**Review by AsdarDevelops — COMMENTED** [review]
Unsure of what difference there is of this entry and the one right above it. Added.

**Review by Adri1 — APPROVED** [review]

**Review by CodeTappert — APPROVED** [review]

**Review by NicusPulcis — APPROVED** [review]
italian is right (after seeing the screenshots)

**Review by josericardo-fo — APPROVED** [review]

**Review by frutescens — APPROVED** [review]

**Review by KimJeongSun — APPROVED** [review]

**Review by KimJeongSun — CHANGES_REQUESTED** [review]
need to resolve conflicts

**Comment by EnochG1** [comment]
@KimJeongSun Done. Please re-review.

**Review by KimJeongSun — APPROVED** [review]

**Pull request merged**